scripts/validate_docs_snippet_line_numbers.py

```diff
@@ -1,205 +1,189 @@
+    parts = raw_path.split("#")
 
-    Args:
-        docusaurus_refs: A map between Python files and the docs that reference them.
-
-    Returns:
-        A list of broken references.
-    """
-    all_broken_refs: List[DocusaurusRef] = []
-    for docs_file, ref_list in docusaurus_refs.items():
-        logger.debug(f"Checking {docs_file} for snippet validity")
-        file_broken_refs: List[DocusaurusRef] = _evaluate_snippet_validity(
-            docs_file, ref_list
-        )
-        all_broken_refs.extend(file_broken_refs)
+    target_path = parts[0]
+    while target_path.startswith("../"):
+        target_path = target_path[3:]
 
-    return all_broken_refs
+    target_lines: Optional[Tuple[int, int]]
+    if len(parts) == 1:
+        target_lines = None
+    else:
+        line_nos = parts[1].split("-")
+
+        start = int(line_nos[0][1:])
+        end: int
+        if len(line_nos) == 1:
+            end = start
+        else:
+            end = int(line_nos[1][1:])
+        target_lines = (start, end)
+
+    return Reference(
+        raw=raw_path,
+        origin_path=file,
+        origin_line=line,
+        target_path=target_path,
+        target_lines=target_lines,
+    )
 
 
-def _evaluate_snippet_validity(
-    file: str, ref_list: List[DocusaurusRef]
-) -> List[DocusaurusRef]:
-    with open(file) as f:
-        file_contents: List[str] = f.readlines()
+def determine_results(refs: List[Reference]) -> List[Result]:
+    all_results = []
+    for ref in refs:
+        result = _determine_result(ref)
+        all_results.append(result)
 
-    broken_refs: List[DocusaurusRef] = []
-    for ref in ref_list:
-        if not _validate_docusaurus_ref(ref, file_contents):
-            broken_refs.append(ref)
+    return all_results
 
-    return broken_refs
 
-
-def _validate_docusaurus_ref(
-    docusaurus_ref: DocusaurusRef, file_contents: List[str]
-) -> bool:
-    start_line, end_line = docusaurus_ref.line_numbers
-    if start_line > len(file_contents) or end_line > len(file_contents):
-        logger.warning(
-            f"Invalid snippet due to out-of-bounds reference: {docusaurus_ref}"
+def _determine_result(ref: Reference) -> Result:
+    if ref.target_lines is None:
+        return Result(
+            ref=ref,
+            status=Status.COMPLETE,
         )
-        return False
-
-    return _validate_opening_tag(start_line, file_contents) and _validate_closing_tag(
-        end_line, file_contents
-    )
-
 
-def _validate_opening_tag(start_line: int, file_contents: List[str]) -> bool:
-    ptr: int = start_line - 1
+    with open(ref.target_path) as f:
+        lines = f.readlines()
+
+    start, end = ref.target_lines
+
+    try:
+        open_tag = lines[start - 2]
+        valid_open = "<snippet" in open_tag
+    except IndexError:
+        valid_open = False
+
+    try:
+        close_tag = lines[end]
+        valid_close = "snippet>" in close_tag
+    except IndexError:
+        valid_close = False
```

Comment:
```
# <snippet> # corresponds to lines[start-2]
...
# </snippet> # corresponds to lines[end]
```

Need to do a bit of messing around to ensure that indexes match with line numbers (since we have 0 vs 1 indexing)